motor-control/firmware/motor_encoder_hardware.c

```diff
@@ -0,0 +1,48 @@
+//Initialize Header Files
+#include "motor_encoder_hardware.h"
+
+#include "FreeRTOS.h"
+#include "task.h"
+
+#include "platform_specific_hal_conf.h"
+
+void motor_encoder_set_pin(void* port, uint16_t pin, uint8_t active_setting)
+{
+    if(!port) {return;}
+    // Hall Function here
```

Comment:
let's avoid comments like this that restate exactly what the next line does or are notes to self. A typical rule is that comments are 
- in the general case, about _why_ you're doing it
- in some cases where the _what_ is weird, the what, but this should be pretty rare (HAL interaction is a pretty normal place to have a "what" comment, like "configure this timer for encoder input per manual section xx.yy.zz"
- in extreme cases, _how_ you're doing something, but usually it's better to figure out a way to do something that doesn't need a how comment in the first place. 

An example is this function: https://github.com/Opentrons/ot3-firmware/blob/4698d1eb0f01f991d476cf8e575e073e4ea7f2c9/bootloader/firmware/updater.c#L190-L219 which is what's responsible for moving from bootloader to application.
- It maybe could use a "why" comment, but given that the function is called `fw_update_start_application` you can kind of see what it does
- the "what" comment about the systick is good because those are kind of cryptic calls; same with the MSP setting
- The assembler snippet has "what" and "why"; it could also have "how", but at that point it would just be restating arm thumb assembler docs since it's literally one assembler directive